tests/gh-actions/install_KinD_create_KinD_cluster_install_kustomize.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+set -e
+
+error_exit() {
+    echo "Error occurred in script at line: ${1}."
+    exit 1
+}
+
+trap 'error_exit $LINENO' ERR
+
+echo "Install KinD..."
+sudo swapoff -a
+
+# This conditional helps running GH Workflows through
+# [act](https://github.com/nektos/act)
+if [ -e /swapfile ]; then
+    sudo rm -f /swapfile
+    sudo mkdir -p /tmp/etcd
+    sudo mount -t tmpfs tmpfs /tmp/etcd
+fi
+
+{
+    curl -Lo ./kind https://kind.sigs.k8s.io/dl/v0.20.0/kind-linux-amd64
+    chmod +x ./kind
+    sudo mv kind /usr/local/bin
+} || { echo "Failed to install KinD"; exit 1; }
+
+
+echo "Creating KinD cluster ..."
+echo "
+apiVersion: kind.x-k8s.io/v1alpha4
+kind: Cluster
+# Configure registry for KinD.
+containerdConfigPatches:
+- |-
+  [plugins.\"io.containerd.grpc.v1.cri\".registry.mirrors.\"$REGISTRY_NAME:$REGISTRY_PORT\"]
+    endpoint = [\"http://$REGISTRY_NAME:$REGISTRY_PORT\"]
```

Comment:
I think we should escape the `$` signs here, otherwise I think the bash script will try to replace it with the corresponding env variable, if I am not wrong. Not sure if that could be realted to the test failures though!

```suggestion
  [plugins.\"io.containerd.grpc.v1.cri\".registry.mirrors.\"\$REGISTRY_NAME:\$REGISTRY_PORT\"]
    endpoint = [\"http://\$REGISTRY_NAME:\$REGISTRY_PORT\"]
```

BTW I'd keep the `kind-cluster` configuration as a separated file to avoid having to escape any double quotes and so on, but that is just my preference :smile:

Reply:
afaik you are right

Reply:
@lampajr the change you suggested in https://github.com/kubeflow/manifests/pull/2805#discussion_r1705838112, does it work when you run the script locally?
Mine fails to write the configuration. It was present as originally in https://github.com/kubeflow/manifests/blob/master/tests/gh-actions/kind-cluster.yaml which might not require changing.
Again, if it works in your local system it's well and good. It also seems to be working here in the CI/CD so it might not be the reason for failing tests.